actors/miner/src/lib.rs

```diff
@@ -3793,9 +3794,20 @@ fn extend_simple_qap_sector(
 
     new_sector.expiration = new_expiration;
     new_sector.power_base_epoch = curr_epoch;
-    if sector.verified_deal_weight > BigInt::zero() {
-        let old_duration = sector.expiration - sector.power_base_epoch;
-        let deal_space = &sector.deal_weight / old_duration;
+    let old_duration = sector.expiration - sector.power_base_epoch;
+    let new_duration = new_sector.expiration - new_sector.power_base_epoch;
+
+    // Update the non-verified deal weights. This won't change power, it'll just keep it the same
+    // relative to the updated power base epoch.
+    if sector.deal_weight.is_positive() {
+        // (old_deal_weight) / old_duration -> old_space
+        // old_space * (old_expiration - curr_epoch) -> remaining spacetime in the deals.
+        new_sector.deal_weight =
```

Comment:
This code preserves the spacetime relative to the new lifetime. Without this change, the deal weight can exceed the sector spacetime (`size * (expiration - power_base_epoch)`). I think the numbers will work out (negative values in https://github.com/filecoin-project/builtin-actors/blob/3dcb014fe6ab06ba97e7fc7f9156c2f4cc2561c2/actors/miner/src/policy.rs#L105-L119 should be fine?). But it still makes me nervous.

Reply:
I see, yeah luckily this works since this case is live on the network today but I agree what you have here is probably better.  This has me more convinced that we should deprecate deal weight soon.